src/main/java/frc/robot/commands/ShooterHeld.java

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.CDSSubsystem;
+import frc.robot.subsystems.LimelightSubsystem;
+import frc.robot.subsystems.ShooterSubsystem;
+
+public class ShooterHeld extends CommandBase {
+  private ShooterSubsystem m_ShooterSubsystem;
+  private LimelightSubsystem m_LimelightSubsystem;
+  private CDSSubsystem m_CDSSubsystem;
+  private int i;
+  private boolean LLEnabled;
+
+  /** Creates a new ShooterHeld. */
+  public ShooterHeld(
+      ShooterSubsystem shooterSubsystem,
+      LimelightSubsystem limelightSubsystem,
+      CDSSubsystem cdsSubsystem,
+      boolean llEnabled) {
+    addRequirements(shooterSubsystem);
+    m_ShooterSubsystem = shooterSubsystem;
+    m_LimelightSubsystem = limelightSubsystem;
+    // m_CDSSubsystem = cdsSubsystem;
+    SmartDashboard.putBoolean("wheelReady", false);
+    LLEnabled = llEnabled;
+    // Use addRequirements() here to declare subsystem dependencies.
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    i = 0;
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    m_ShooterSubsystem.prime();
+    if (m_ShooterSubsystem.wheelReady()) {
+      if (i > 0 || !LLEnabled || m_LimelightSubsystem.calculatePID() == 0.0) {
+        i++;
+        m_ShooterSubsystem.runCargo(true, true);
+        m_ShooterSubsystem.setCargoBoolean(true);
+        if (i >= 50) {// 1000 miliseconds delay
```

Comment:
If the delay is here to make sure we've shot all of the cargo in the robot, we can potentially poll our CDS sensors to see if they're all clear and finish the command more precisely

Reply:
True, it's being used as a placeholder for the moment as I don't think the CDS has beam break functionality yet? Correct me if I'm wrong?

Reply:
Should we maybe open an issue so we can add the functionality later?

Reply:
Sounds good! Yes my understanding is they don't expose a method for sensor status yet (do those operations internally) so an issue filed to them and a TODO comment here would be good enough

Reply:
Issue created #105